File: android/src/main/java/org/ligi/passandroid/ui/pass_view_holder/CondensedPassViewHolder.java
package org.ligi.passandroid.ui.pass_view_holder;

import android.app.Activity;
import android.view.View;
import org.ligi.passandroid.model.PassStore;
import org.ligi.passandroid.model.pass.Pass;
import static android.view.View.GONE;
import static android.view.View.VISIBLE;

public class CondensedPassViewHolder extends PassViewHolder {

    public CondensedPassViewHolder(View view) {
        super(view);
    }

    @Override
    public void apply(Pass pass, PassStore passStore, Activity activity) {
        super.apply(pass, passStore, activity);

        final String extraString = getExtraString();
        if (extraString != null && !extraString.isEmpty()) {
            dateOrExtraText.setText(extraString);
            dateOrExtraText.setVisibility(VISIBLE);
        } else {
            dateOrExtraText.setVisibility(GONE);
        }

        addCalendar.setText(getTimeInfoString());

    }
}

File: android/src/main/java/org/ligi/passandroid/ui/pass_view_holder/CondensedPassViewHolder.kt
package org.ligi.passandroid.ui.pass_view_holder

import android.app.Activity
import androidx.cardview.widget.CardView
import android.view.View
import android.widget.TextView
import org.ligi.passandroid.R
import org.ligi.passandroid.model.PassStore
import org.ligi.passandroid.model.pass.Pass

class CondensedPassViewHolder(view: CardView) : PassViewHolder(view) {

    override fun apply(pass: Pass, passStore: PassStore, activity: Activity) {
        super.apply(pass, passStore, activity)

        val extraString = getExtraString(pass)

        val date = view.findViewById<TextView>(R.id.date)


        if (extraString.isNullOrBlank()) {
            date.visibility = View.GONE
        } else {
            date.text = extraString
            date.visibility = View.VISIBLE
        }

        view.findViewById<TextView>(R.id.timeButton).text = getTimeInfoString(pass)

    }
}
